src/library.js

```diff
@@ -3329,8 +3329,17 @@ LibraryManager.library = {
     });
   },
 
-#if DYNCALLS || !WASM_BIGINT
-  $dynCallLegacy: function(sig, ptr, args) {
+  /* Different functions related to calling or binding a function pointer in Wasm from JS:
+    1. _dynCallI32Pair(): internal, do not call in user code.
+    2. dynCall(sig, ptr, argsArray): public user facing function, dynamic signature
+         dispatch with i64s as a i32 pair.
+    3. dynCall_xxx(ptr, arg0, arg1, ...): same as above, i64s as i32 pair, but with static dispatch.
+    4. getDynCaller(sig, ptr): public user facting function, binds to a function using dynamic
+         signature dispatch with i64s as a i32 pair.
+    5. wbind(ptr): binds to a function with i64s as BigInts calling convention.
+    6. wbindArray(ptr): same as above, but returned function should be invoked with args passed in an array.
```

Comment:
I guess this means that wbind cannot be used (at least not in the general case) unless WASM_BIGINT is enabled?
I guess this inherently the case (and we can't offer, for example, wbind32 for such users) because wbind always binds directly the the compiler output whereas dynCall can use binaryen-generated stubs split i64s?

Reply:
> I guess this means that wbind cannot be used (at least not in the general case) unless WASM_BIGINT is enabled?

That is right: wbind can be used for non-64-bit signatures in all builds, but only for 64-bit signatures if WASM_BIGINT is enabled.

That might at first sound like a drawback, but in my mind that is explicitly a desirable property. That is because the i32pairs vs BigInt ABI abstraction is not a hidden internal detail, but the caller must be ready to handle the calling convention and rewrite their code to adjust to the right calling convention.

The scenarios where a caller is an agnostic "passer-through" seem to be a very rare and quirky behavior. Such callers would be expected to have to manually write
```js
  function passesThroughIncomingArgsToAWasmFunctionPtr(funcPtr, args) {
#if WASM_BIGINT
    wbindArray(funcPtr)(args);
#else
    dynCall(getTheSigSomewhere, funcPtr, args);
#endif
  }
```
but I can't really figure out what kind of usage scenario ever wants to do that. The only two cases we currently have of this are
1) the invoke functions: there like you note below, it is actually an internal implementation detail that it just passes args through,
2) embind's `$embind__requireFunction`. Here I think it is actually a bug/bad behavior that it does this, since this means that when you flip between `WASM_BIGINT` enabled vs disabled, it will change the 64-bit ABI for all embind using code, leading to the "all caller code sites need to migrate in one go" problem. Maybe for all current embind users this will not be a problem in practice(?) - if it is used only by smaller authors that control all their JS library code, then it is probably not a problem. (We don't use embind at Unity so we don't care about this)

We could create a function
```js
function wbindFlexible64BitCallingConvention(sig, ptr, args) {
#if WASM_BIGINT
  return wbindArray(funcPtr)(args);
#else
  return dynCall(getTheSigSomewhere, funcPtr, args);
#endif
}
```
to wrap this to caller, but I don't see there being any sensible uses of this, so better to just explicitly ifdef the few places where it is a valid thing to do. (== case (1) above)